src/irr/asset/CGraphicsPipelineLoaderMTL.cpp

```diff
@@ -79,30 +462,99 @@ core::smart_refctd_ptr<ICPUPipelineLayout> CGraphicsPipelineLoaderMTL::makePipel
 
 SAssetBundle CGraphicsPipelineLoaderMTL::loadAsset(io::IReadFile* _file, const IAssetLoader::SAssetLoadParams& _params, IAssetLoader::IAssetLoaderOverride* _override, uint32_t _hierarchyLevel)
 {
+    constexpr uint32_t POSITION = 0u;
+    constexpr uint32_t UV = 2u;
+    constexpr uint32_t NORMAL = 3u;
+    constexpr uint32_t BND_NUM = 0u;
+
+    SContext ctx(
+        asset::IAssetLoader::SAssetLoadContext{
+            _params,
+            _file
+        },
+        _hierarchyLevel,
+        _override
+    );
+    const io::path fullName = _file->getFileName();
+	const std::string relPath = (io::IFileSystem::getFileDir(fullName)+"/").c_str();
+
     auto materials = readMaterials(_file);
 
-    core::vector<core::smart_refctd_ptr<ICPURenderpassIndependentPipeline>> pipelines(materials.size());
-    SVertexInputParams vtxParams;
-    SBlendParams blendParams;
-    SPrimitiveAssemblyParams primParams;
-    SRasterizationParams rasterParams;
-    for (size_t i = 0ull; i < pipelines.size(); ++i)
+    constexpr uint32_t PIPELINE_PERMUTATION_COUNT = 2u;
+
+    core::vector<core::smart_refctd_ptr<ICPURenderpassIndependentPipeline>> pipelines(materials.size()*PIPELINE_PERMUTATION_COUNT);
+    for (size_t i = 0ull; i < materials.size(); ++i)
     {
-        auto layout = makePipelineLayoutFromMtl(materials[i]);
-        pipelines[i] = core::make_smart_refctd_ptr<ICPURenderpassIndependentPipeline>(nullptr, std::move(layout), nullptr, nullptr, vtxParams, blendParams, primParams, rasterParams);
+        SVertexInputParams vtxParams;
+        SBlendParams blendParams;
+        SPrimitiveAssemblyParams primParams;
+        SRasterizationParams rasterParams;
+
         if (materials[i].maps[CMTLPipelineMetadata::SMtl::EMP_OPACITY].size())
         {
-            for (uint32_t i = 0u; i < SBlendParams::MAX_COLOR_ATTACHMENT_COUNT; ++i)
+            for (uint32_t k = 0u; k < SBlendParams::MAX_COLOR_ATTACHMENT_COUNT; ++k)
```

Comment:
still issue of why are you setting blending on all attachments!?

Reply:
idk, seemed to me like a reasoneable default also for other attachments (i.e. blend params for all attachments are the same by default)

Reply:
don't enable blends on attachments that you don't need to (don't confuse the driver).